### PR TITLE
Renderer refactor 2

### PR DESCRIPTION
**Pull request by Arcnor**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [x] Build and Dependency changes
- [x] Runtime changes
  - [x] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [ ] Other changes

### Description
More renderer refactor, this time abstracting the triangle rendering code & making it so a single vertex stream is used.

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have documented any new or modified functionality.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

File: renderer/HardwareBaseGPU.cpp [hunk 1]
```diff
@@ -434,3 +438,319 @@ void rend_GetStatistics(tRendererStats *stats) {
     memset(stats, 0, sizeof(tRendererStats));
   }
 }
+
+#ifdef DEDICATED_ONLY
```
Comment (Lgt2x) [inline renderer/HardwareBaseGPU.cpp:442]: Never defined anywhere by the way

File: renderer/HardwareBaseGPU.cpp [hunk 2]
```diff
@@ -434,3 +438,319 @@ void rend_GetStatistics(tRendererStats *stats) {
...
+    vData->uv1.s = pnt->p3_u2 * xscalar;
+    vData->uv1.t = pnt->p3_v2 * yscalar;
+    vData->uv1.r = 0.0f;
+    vData->uv1.w = 1.0f;
+
+    // Finally, specify a vertex
+    /*
+    vData->pos.x=pnt->p3_sx+x_add;
+    vData->pos.y=pnt->p3_sy+y_add;
+    vData->pos.z = -std::max(0,std::min(1.0,1.0-(1.0/(pnt->p3_z+Z_bias))));
```
Comment (Lgt2x) [inline renderer/HardwareBaseGPU.cpp:742]: not in the original function?

**Review by Lgt2x — APPROVED** [review]

File: renderer/HardwareBaseGPU.cpp (hunk 2 above)
Comment (Arcnor) [reply renderer/HardwareBaseGPU.cpp:742]: Yeah, this is from the original, I didn't add any commented code on purpose AFAIK.

**Comment by JeodC** [comment]
I get a hard crash on windows with this. Last good commit is the one right before it https://github.com/DescentDevelopers/Descent3/commit/1f94e623b24ef7bb5a1e4bc797842b2d806e70e3

**Pull request merged**